src/services/recentlyViewedProjects.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+
+angular.module("openshiftCommonServices")
+  .service("RecentlyViewedProjectsService", function($filter){
+
+    var recentlyViewedProjsKey = "openshift/recently-viewed-project-uids";
+
+    var getProjectUIDs = function() {
+      var recentlyViewed = localStorage.getItem(recentlyViewedProjsKey);
+      return recentlyViewed ? JSON.parse(recentlyViewed) : [];
+    };
+
+    var addProjectUID = function(uid) {
+      var recentlyViewed = getProjectUIDs();
+
+      // add to front of list
+      recentlyViewed.unshift(uid);
+
+      // no dups
+      recentlyViewed = _.uniq(recentlyViewed);
+
+      // limit to 5 items
+      recentlyViewed = _.take(recentlyViewed, 5);
+
+      setRecentlyViewedProjects(recentlyViewed);
+    };
+
+    var clear = function() {
+      localStorage.removeItem(recentlyViewedProjsKey);
+    };
+
+    var setRecentlyViewedProjects = function(recentlyViewed) {
+      localStorage.setItem(recentlyViewedProjsKey, JSON.stringify(recentlyViewed));
+    };
+
+    var orderByMostRecentlyViewed = function(projects) {
+      var recentlyViewedProjects = [];
+      var recentlyViewedIds = getProjectUIDs();
+
+      // first order by creationDate
+      projects =  $filter('orderObjectsByDate')(projects, true);
```

Comment:
Suggest we move this below the `_.each` so that we sort the smaller array after the remove the other recently-viewed projects.